src/vs/editor/common/model/textModel.ts

```diff
@@ -2413,6 +2454,8 @@ export class ModelDecorationOptions implements model.IModelDecorationOptions {
 		this.before = options.before ? ModelDecorationInjectedTextOptions.from(options.before) : null;
 		this.hideInCommentTokens = options.hideInCommentTokens ?? false;
 		this.hideInStringTokens = options.hideInStringTokens ?? false;
+		this.affectsFont = !!options.fontSize || !!options.fontFamily || !!options.fontWeight || !!options.fontStyle;
+		this.stickiness = this.affectsFont ? model.TrackedRangeStickiness.NeverGrowsWhenTypingAtEdges : (options.stickiness || model.TrackedRangeStickiness.AlwaysGrowsWhenTypingAtEdges);
```

Comment:
stickiness is something configured independently of font size. It should not be forced to be NeverGrowsWhenTypingAtEdges when font size is configured, the creator of the decoration should decide what they want.

Reply:
Thanks will do!